File: 0206-reverse-linked-list/0206-reverse-linked-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* reverseList(ListNode* head) {
        ListNode* p = nullptr;
        ListNode* q = nullptr;
        ListNode* first = head;
        while(first!= nullptr){
            p = q;
            q = first;
            first = first->next;
            q->next = p;
        }
        head = q;
        return head;
    }
}; 

File: 0206-reverse-linked-list/0206-reverse-linked-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def reverseList(self, head):
        """
        :type head: Optional[ListNode]
        :rtype: Optional[ListNode]
        """
        curr = head
        p = None
        q = None
        
        while(curr != None):
            p = q
            q = curr
            curr = curr.next
            q.next = p
        
        head = q
        return head
        